### DISCUSSION TITLE
Push is rejected

### DISCUSSION THREAD

**Discussion by areichPuC** [1DSAG/UI5-Best-Practice#24] [General]
Hello :)

I need some help for my first push. It is currently being rejected:
`error: failed to push some refs to 'https://github.com/marianfoo/UI5-Best-Practice.git'`
`To https://github.com/marianfoo/UI5-Best-Practice.git`
`!	refs/heads/i18nAnnotations:refs/heads/i18nAnnotations	[remote rejected] (permission denied)`

Here is what I have done:
1. cloned fork https://github.com/marianfoo/UI5-Best-Practice/tree/i18n-improve
2. created a local branch (name: i18nAnnotations)
3. added a new image and edited index.md. In addition, Gemfile and Gemfile.lock were automatically expanded, because I have to add webrick under Windows.
4. everything committed with the message: 
`feat(i18n): annotations`
`added text for i18n annotations in index.md`

5. Tried `git push`, but it was rejected.

Oh, and I created a token on GitHub with the categories gist, read:org and repo.

Does anyone have an idea why my push is being rejected?

Thanks a lot and best regards
Anne

**Comment by vobu** [comment]
hi, you're trying to push against @marianfoo's fork of the repo - given that it's his private one, you get the forbidden error.
instead, fork the original repo https://github.com/1DSAG/UI5-Best-Practice into your account and operate on that aka your personal fork. 
hth, v.

**Reply by vobu** [reply]
during today's meeting, the exact same challenge arose, and I think I misunderstood your original question: you wanted to contribute to a _pull request_ (as required in https://github.com/1DSAG/UI5-Best-Practice/discussions/15), not to a repo or fork.
so here we go: in order to contribute to a PR, you need to
- clone ("check out") the PR itself (which -technically- is just a git branch), e.g. via the cli `gh pr checkout 11` or via somehow via an IDE's mechanism
- make updates on your clone + `git commit`
- `git push` to the PR, meaning to a `remote` pointing to the PR. Usually this `remote` is already set automatically when you clone the PR.
- voilà: you'll see the commit to the PR on the commit history of PR page, e.g. on https://github.com/1DSAG/UI5-Best-Practice/pull/11

hope that really helps now,
v.

**Reply by areichPuC** [reply]
Something still hasn't worked. I was able to do a git push, but I don't see anything in the commit history. Maybe we can look at it tomorrow? I'll be there from 09:30.

**Reply by vobu** [reply]
hey anne, won't be able to make it tomorrow. in case the people round tomorrow doesn't get this figured out, ping me directly. then we'll sort through this in a quick call.
happy hunting, v.
